frontend/src/screens/subwallets/SubwalletList.tsx

```diff
@@ -0,0 +1,233 @@
+import Loading from "src/components/Loading";
+import { Button } from "src/components/ui/button";
+import { Input } from "src/components/ui/input";
+import { Label } from "src/components/ui/label";
+import { LoadingButton } from "src/components/ui/loading-button";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from "src/components/ui/tooltip";
+import { useToast } from "src/components/ui/use-toast";
+import UpgradeCard from "src/components/UpgradeCard";
+import { UpgradeDialog } from "src/components/UpgradeDialog";
+import { useAlbyMe } from "src/hooks/useAlbyMe";
+import { useApps } from "src/hooks/useApps";
+import { useInfo } from "src/hooks/useInfo";
+import { createApp } from "src/requests/createApp";
+import { CreateAppRequest } from "src/types";
+import { handleRequestError } from "src/utils/handleRequestError";
+
+export function SubwalletList() {
+  const navigate = useNavigate();
+  const [name, setName] = React.useState("");
+  const { data: apps } = useApps();
+  const { toast } = useToast();
+  const { data: albyMe, error: albyMeError } = useAlbyMe();
+  const { data: info } = useInfo();
+  const onboardedApps = apps
+    ?.filter((app) => app.metadata?.app_store_app_id === "uncle-jim")
+    .sort((a, b) => (a.createdAt < b.createdAt ? 1 : -1));
+
+  const subwalletAmount = onboardedApps?.reduce(
+    (total, app) => total + app.balance,
+    0
+  );
+
+  const [isLoading, setLoading] = React.useState(false);
+  const [showIntro, setShowIntro] = React.useState(true);
+  const showForm =
+    albyMe?.subscription.plan_code ||
+    (onboardedApps && onboardedApps?.length < 3);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setLoading(true);
+
+    try {
+      const createAppRequest: CreateAppRequest = {
+        name,
+        scopes: [
+          "get_balance",
+          "get_info",
+          "list_transactions",
+          "lookup_invoice",
+          "make_invoice",
+          "notifications",
+          "pay_invoice",
+        ],
+        isolated: true,
+        metadata: {
+          app_store_app_id: "uncle-jim",
+        },
+      };
+
+      const createAppResponse = await createApp(createAppRequest);
+
+      navigate(`/sub-wallets/created`, {
+        state: createAppResponse,
+      });
+
+      toast({ title: "New sub-wallet created for " + name });
+    } catch (error) {
+      handleRequestError(toast, "Failed to create app", error);
+    }
+    setLoading(false);
+  };
+
+  if (!info || (info.albyAccountConnected && !albyMe && !albyMeError)) {
+    // make sure to not render the incorrect component
+    return <Loading />;
+  }
+
+  return (
+    <div className="grid gap-5">
+      <AppHeader
+        title="Sub-wallets"
+        contentRight={
+          <>
+            {subwalletAmount !== undefined && (
+              <Tooltip>
+                <TooltipTrigger>
+                  <Button variant="outline">
```

Comment:
This is a button but doesn't seem to have an action?

Reply:
It also breaks on mobile cc @stackingsaunter need your input here

<img width="312" alt="Screenshot 2025-04-10 at 5 39 49 PM" src="https://github.com/user-attachments/assets/297411df-2a77-4aab-852f-3c58692f5b47" />

Reply:
Let's just hide this button on xs.